managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -443,7 +443,8 @@ public void asyncOpenReadOnlyManagedLedger(String managedLedgerName,
 
         }).exceptionally(e -> {
             log.error("[{}] Failed to initialize Read-only managed ledger", managedLedgerName, e);
-            callback.openReadOnlyManagedLedgerFailed((ManagedLedgerException) e.getCause(), ctx);
```

Comment:
Why need this change? Will this cause class cast exception?

Reply:
Yes, because this is a PluginClassLoader, it did not directly throw the `ClassCastException`.
<img width="2483" alt="image" src="https://github.com/apache/pulsar/assets/15684156/0973d65a-a8e9-4cb6-baaf-57b7019dedad">
<img width="798" alt="image" src="https://github.com/apache/pulsar/assets/15684156/ef80aa6c-302e-4331-8980-3709e500c7da">